clap_complete/src/dynamic/completer.rs

```diff
@@ -708,3 +723,69 @@ impl CompletionCandidate {
         self.visible
     }
 }
+
+/// This trait is used to provide users a way to add custom value hint to the argument.
+/// This is useful when predefined value hints are not enough.
```

Comment:
The "short" description in docs should get to the point quickly for people to quickly find things

> User-provided completion candidates for an argument
>
> This is useful when predefined value hints are not enough.

(note the need for a blank line because markdown will join them)